src/averageAges.js

```diff
@@ -38,6 +48,18 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const mothers = people.map((person) => person.mother);
+
+  const ages = withChildren
+    ? people.filter((person) => mothers.includes(person.name)
+      && person.sex === 'f')
+      .map((person) => person.died - person.born)
+    : people.filter((person) => person.sex === 'f')
+      .map((person) => person.died - person.born);
+
+  const averageAges = ages.reduce((prev, item) => prev + item, 0) / ages.length;
```

Comment:
I meant, create separate function outside this one and call it in all 3 functions